### PR TITLE
Created basic ball controls/physics

### PR DESCRIPTION
**Pull request by WalterTrustam**
Changes made:
- Added physics material BounceMat to Player aka ball to make it more bouncy
- Added DirectionIndicator object as child of Player to indicate which way the ball is 'facing'
- Modified Angular Drag on Player object to make it come to a stop quicker
- Modified PlayerController.cs for basic ball controls
- Modified camera positioning for more appropriate golf view, just for testing

Controls:
- Left/Right key - Rotates the ball on the horizontal axis
- Up/Down key - Rotates the ball on the vertical axis (aka changing the pitch)
- Space bar down - Starts charging power
- Space bar up - Fires ball in direction it is 'facing' with charged power
- R - for debugging, resets the level

Ball will reset its orientation once it has stopped moving and can then be fired again.

### CODE REVIEW

File: SEM-lab1/Assets/Scripts/PlayerController.cs [hunk 1]
```diff
@@ -1,34 +1,79 @@
...
+    private bool _isCharging = false;       
+    private bool _spaceUp = true;
+    private bool _fired = false;
+    private Quaternion _originalRotation;
 
     // Start is called before the first frame update
     void Start()
     {
         _gameController = GameObject.FindGameObjectWithTag("GameController").GetComponent<GameController>();
+        this._originalRotation = this.transform.rotation; //saves initial rotation for refiring so that the ball is not at a strange angle due to rolling
```
Comment (AlexanderNicholls) [inline SEM-lab1/Assets/Scripts/PlayerController.cs]: Shouldn't need to write "this.X", reference to the gameObject should be implicit.

File: SEM-lab1/Assets/Scripts/PlayerController.cs [hunk 2]
```diff
@@ -1,34 +1,79 @@
...
+
+        if (_isCharging)
+        {
+            _chargeTimeCurr += Time.deltaTime; //increases charge over time, when space is pressed
+        }
 
         Rigidbody r = GetComponent<Rigidbody>();
-        r.velocity = new Vector3(horizontalMovement * speed, 0.0f, verticalMovement * speed);
+        float movementSpeed = r.velocity.magnitude;
+        if (movementSpeed < 0.05 && _fired) //checks if the ball is still moving, ie has a magnitude
```
Comment (AlexanderNicholls) [inline SEM-lab1/Assets/Scripts/PlayerController.cs:59]: Why 0.05 for the movementSpeed?

**Review by AlexanderNicholls — COMMENTED** [review]
Enquiring into the usage of 'this.X' and reasoning behind magic numbers.

File: SEM-lab1/Assets/Scripts/PlayerController.cs (hunk 2 above)
Comment (WalterTrustam) [reply SEM-lab1/Assets/Scripts/PlayerController.cs:59]: Mostly derived from trial and error. If movementSpeed == 0 then the ball appears to have stopped long before it has 'technically' stopped which could be frustrating for the player. The actual value is subject to change depending on feedback.

File: SEM-lab1/Assets/Scripts/PlayerController.cs (hunk 1 above)
Comment (WalterTrustam) [reply SEM-lab1/Assets/Scripts/PlayerController.cs]: Confirmed it works without 'this', will change for the next commit.

**Review by AlexanderNicholls — APPROVED** [review]

**Review by AlexanderNicholls — APPROVED** [review]

**Pull request merged**